pyccel/ast/variable.py

```diff
@@ -823,3 +825,12 @@ def lhs(self):
         The lhs is a
         """
         return self._lhs
+
+    def __eq__(self, other):
+        if isinstance(other, DottedVariable):
+            return self.name == other.name and self.lhs == other.lhs
+
+        return False
+
+    def __hash__(self):
+        return hash(type(self).__name__+ self.name + str(self.lhs))
```

Comment:
I don't like this. `hash` should include all the information necessary to differentiate 2 objects. `str` is a lot less strict and is just there to give a legible representation of a function. Imagine the case where the `lhs` is a function call, it may not be possible to get enough nuance just from the string representation. I would prefer:
```
return hash(type(self).__name__+ self.name) + hash(self.lhs)
```
(hash returns an int so they can be added)

Reply:
or you just can do , `hash((type(self).__name__, self.name, self.lhs))`